pydrawise/client.py

```diff
@@ -370,3 +356,101 @@ async def get_watering_report(
                 entries,
             )
         )
+
+    async def get_watering_report(
+        self, controller: Controller, start: datetime, end: datetime
+    ) -> list[WateringReportEntry]:
+        """Retrieves a watering report for the given controller and time period.
+
+        :param controller: The controller whose watering report to generate.
+        :param start: Start time.
+        :param end: End time."""
+        selector = self._schema.Query.controller(controllerId=controller.id).select(
+            self._schema.Controller.reports.select(
+                self._schema.Reports.watering(
+                    **{
```

Comment:
You can just pass `from` and `until` directly as arguments instead of doing the dict expansion here.

Reply:
This doesn't work because "from" is a reserved keyword in Python.